convert.pl

```diff
@@ -29,16 +30,18 @@ sub uint_from_bits
     return 'uint16_t' if ($bits <= 16);
     return 'uint32_t' if ($bits <= 32);
     return 'uint64_t' if ($bits <= 64);
-    die 'Value too large';
+    confess 'Value too large for bits: '. $bits;
 }
 
 use constant BITS_IN_BYTE => 8;
 
 use constant EXPRESSIONS => {
     'A'                     => ['uint8_t',   8, '$id',           1],
     'A x 0.1'               => ['uint8_t',   8, '$id',         0.1],
+    'A x 100'               => ['uint8_t',   8, '$id',         100],
     'A.0 + B.1'             => ['uint16_t', 16, '$id',           1],
     '(A.1 + B.0) x 8'       => ['uint16_t', 16, 'ntohs($id)',    8],
+    '(A.1 + B.0) x 10'      => ['uint16_t', 16, 'ntohs($id)',   10],
```

Comment:
Out of interest (and so I can reference it in the commit message for these additions), which of the config xml files did you use for your boiler?

Reply:
`perl convert.pl ~/remeha/config/AvantaV1_P5.xml ~/remeha/language/language-en.xml parameter` and `perl convert.pl ~/remeha/config/AvantaV1_P5.xml ~/remeha/language/language-en.xml counter` needed them.
I just noticed that 'counter' would also need "(A.1 + B.0) x 1" and "(A.1 + B.0) x 2". Will update with a PR when I have hacked a bit more.